### PR TITLE
Display board with position absolute

### PR DESCRIPTION
**Pull request by arimet**
Changing the display of a game board. 
We go from flex to a positioning by the absolute

image

### CODE REVIEW

File: src/GameSelector/UserGame.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import { Pressable } from 'react-native';
+import { Board } from '../GameState/Board';
+import { Game } from '../interface';
+
+interface UserGameProps {
+  navigateToGameState: (id: number | undefined) => void;
+  game: Game | undefined;
+}
+
+const GRADIENT_COLORS: any = [];
```
Comment (Aherbeth) [inline src/GameSelector/UserGame.tsx]: I think we can move it to the constants file

File: src/GameState/Board.tsx [hunk 2]
```diff
@@ -1,43 +1,68 @@
...
+        const value = col;
+        result.push({ x, y, value });
+      });
+    });
+
+    return result;
+  };
+  const boardCoordinate = convertBoardToPosition();
+  const boxMarbleSize = 100 / board.length;
+  const boxMarbleSizePourcent = `${100 / board.length}%`;
```
Comment (Aherbeth) [inline src/GameState/Board.tsx]: eventually
```Ts
  const boxMarbleSizePourcent = `${boxMarbleSize}%`;
  ```

File: src/GameState/Board.tsx [hunk 3]
```diff
@@ -1,48 +1,56 @@
...
 import React from 'react';
-import { BOARD_GREEN, BOARD_GREY } from './boardColors';
+import { BOARD_GREEN } from './boardColors';
+import { convertBoardToBoardCoordinate } from './convertBoardToBoardCoordinate';
 import { Marble } from './Marble';
 
 export const Board = (props: any) => {
   const board = props.board;
-  const boardSize = props.preview ? 4 : 12;
+  const boardSize = props.preview ? 32 : '360px';
```
Comment (Luwangel) [inline src/GameState/Board.tsx]: What's the difference between 32 and 360px?

File: src/GameState/convertBoardToBoardCoordinate.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+export type BoardCoordinate = { x: number; y: number; value: number };
+
+export const convertBoardToBoardCoordinate = (board: [][]) => {
+  let result: BoardCoordinate[] = [];
+
+  board.map((row: [], index: number) => {
```
Comment (Luwangel) [inline src/GameState/convertBoardToBoardCoordinate.tsx]: Why do you use a map if you don't return anything?

Comment (Luwangel) [reply src/GameState/convertBoardToBoardCoordinate.tsx]: I think it's better to use two for loops. Or array.reduce combined with a forEach.

**Review by Luwangel — CHANGES_REQUESTED** [review]

File: src/GameState/Board.tsx (hunk 3 above)
Comment (arimet) [reply src/GameState/Board.tsx]: Basically we used the sizes provided by Native Base. 
But it wasn't enough for us. So I unified in px

File: src/GameState/Board.tsx [hunk 5]
```diff
@@ -1,48 +1,64 @@
...
+import { BOARD_GREEN } from './boardColors';
+import { convertBoardToBoardCoordinate } from './convertBoardToBoardCoordinate';
 import { Marble } from './Marble';
 
 export const Board = (props: any) => {
   const board = props.board;
-  const boardSize = props.preview ? 4 : 12;
-  const marbleColorSize = props.preview ? 4 : 12;
-  const marbleEmptySize = props.preview ? 2 : 4;
+  const boardSize = props.preview ? '128px' : '364px';
```
Comment (Aherbeth) [inline src/GameState/Board.tsx:10]: maybe we can have sizes in a separate file, in order to make this more "readable"

File: src/GameState/Board.tsx [hunk 6]
```diff
@@ -1,48 +1,64 @@
...
-                marbleClickedCoordinates?.y === rowIndex
-                  ? BOARD_GREY
-                  : BOARD_GREEN
-              }
-              m={0}
-              p={0}
-            >
+    <View width={boardSize} height={boardSize}>
+      <ImageBackground
+        source={require('./board.png')}
```
Comment (Aherbeth) [inline src/GameState/Board.tsx:24]: Maybe import it ?

**Review by Aherbeth — APPROVED** [review]

File: src/GameState/Board.tsx (hunk 6 above)
Comment (arimet) [reply src/GameState/Board.tsx:24]: This modification has been realized in the following PR

File: src/GameState/Board.tsx (hunk 5 above)
Comment (arimet) [reply src/GameState/Board.tsx:10]: This modification has been realized in the following PR

File: src/GameState/Board.tsx [hunk 7]
```diff
@@ -1,48 +1,64 @@
...
+import { convertBoardToBoardCoordinate } from './convertBoardToBoardCoordinate';
 import { Marble } from './Marble';
 
 export const Board = (props: any) => {
   const board = props.board;
-  const boardSize = props.preview ? 4 : 12;
-  const marbleColorSize = props.preview ? 4 : 12;
-  const marbleEmptySize = props.preview ? 2 : 4;
+  const boardSize = props.preview ? '128px' : '364px';
+  const marbleColorSize = props.preview ? '16px' : '48px';
```
Comment (Luwangel) [inline src/GameState/Board.tsx:11]: We usually use constants. But don't change it now.

File: src/GameState/Board.tsx (hunk 6 above)
Comment (Luwangel) [inline src/GameState/Board.tsx:24]: Could you import it on top of the file?

File: src/GameState/Marble.tsx [hunk 8]
```diff
@@ -4,28 +4,29 @@ import React from 'react';
...
-  const child = <Circle size={size} m={0} p={0} bgColor={getMarbleColor(props.value)} />;
+  const child = (
+    <Circle size={size} m={0} p={0} bgColor={getMarbleColor(props.value)} />
+  );
 
   const marbleClick = async () => {
-    await props.setMarbleClicked({ x: props.cellIndex, y: props.rowIndex, value: props.value, isExit: false })
-    props.setMarbleClickedCoordinates({ x: props.cellIndex, y: props.rowIndex })
-  }
+    await props.setMarbleClicked({
```
Comment (Luwangel) [inline src/GameState/Marble.tsx]: It's difficult to understand why this method is async and why the next one is not. The naming is not good IMO.

File: src/GameState/Marble.tsx [hunk 9]
```diff
@@ -4,28 +4,29 @@ import React from 'react';
...
 export const Marble = (props: any) => {
-
-
   const size = props.size;
-  const child = <Circle size={size} m={0} p={0} bgColor={getMarbleColor(props.value)} />;
+  const child = (
+    <Circle size={size} m={0} p={0} bgColor={getMarbleColor(props.value)} />
+  );
 
   const marbleClick = async () => {
```
Comment (Luwangel) [inline src/GameState/Marble.tsx]: We generally use the keyword handle: `handleClickMarble` when the method is linked to an event.

File: src/GameState/Board.tsx (hunk 7 above)
Comment (arimet) [reply src/GameState/Board.tsx:11]: It's done in PR #15

File: src/GameState/Board.tsx (hunk 6 above)
Comment (arimet) [reply src/GameState/Board.tsx:24]: It's done in PR #15

File: src/GameState/Marble.tsx (hunk 8 above)
Comment (arimet) [reply src/GameState/Marble.tsx]: There was no reason, I removed it

**Review by Luwangel — APPROVED** [review]

**Pull request merged**